File: backend/domain/conversation.py
from datetime import datetime
from pydantic import Field
from .common import TimestampedModel
from .enums import ConversationInputType, Speaker


class ConversationTurn(TimestampedModel):
    turn_id: str
    session_id: str
    speaker: Speaker
    input_type: ConversationInputType
    content: str | None = None
    language: str | None = None
    media_reference: str | None = None
    created_at: datetime = Field(
        default_factor=lambda: datetime.now().astimezone())

File: patient-web/src/api/conversation.ts
import { apiRequest } from "./client";

export type ConversationInputType = "AUDIO" | "GUIDED_INPUT" | "TEXT";

export interface ConversationTurn {
  turn_id: string;
  session_id: string;
  speaker: string;
  input_type: ConversationInputType;
  content: string | null;
  language: string | null;
  media_reference: string | null;
  created_at: string;
}

interface ConversationTurnResponse {
  data: ConversationTurn;
}

export async function submitConversationTurn(
  sessionId: string,
  inputType: ConversationInputType,
  content: string,
  language: string,
): Promise<ConversationTurn> {
  const response = await apiRequest<ConversationTurnResponse>(
    `/sessions/${encodeURIComponent(sessionId)}/conversation/turns`,
    {
      method: "POST",
      body: JSON.stringify({
        input_type: inputType,
        content,
        language,
      }),
    },
  );

  return response.data;
}
